drf_util/views.py

```diff
@@ -111,16 +169,55 @@ def get_query_serializer(self):
     def get_serializer_create_class(self):
         return self.get_serializer_by_action() or self.serializer_create_class or self.serializer_class
 
+    def get_serializer_retrieve_class(self):
+        return self.get_serializer_by_action() or self.serializer_retrieve_class or self.serializer_class
+
     def get_object_id(self):
         return self.kwargs.get(self.lookup_field)
 
 
-class BaseListModelMixin(mixins.ListModelMixin):
+class BaseListModelMixin:
     filter_class = None
     search_fields = ()
+    filterset_fields = []
     ordering_fields = '__all__'
     ordering = ['-id']
 
+    """
+    List a queryset.
+    """
+
+    def __init__(self, **kwargs):
+        is_filter_class_set = False
+
+        # Class has filter_class set , check if value is not None
+        if hasattr(self, 'filter_class') and getattr(self, 'filter_class', None):
+            is_filter_class_set = True
+
+        # Automatically create a filter class for available filter set fields
+        if not is_filter_class_set and hasattr(self, 'filterset_fields'):
+            # Dynamic filter classes
+            namespace = self.queryset.model.__name__  # noqa
+            self.filter_class = type(f'{namespace}FilterClass', (FilterSet,), {
+                'Meta': type(f'{namespace}FilterMetaClass', (object,), {
+                    'model': self.queryset.model,  # noqa
+                    'fields': self.filterset_fields
+                })
+            })
+
+        super().__init__(**kwargs)
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())  # noqa
+
+        page = self.paginate_queryset(queryset)  # noqa
+        if page is not None:
+            serializer = self.get_serializer_retrieve(page, many=True)  # noqa
+            return self.get_paginated_response(serializer.data)  # noqa
+
+        serializer = self.get_serializer_retrieve(queryset, many=True)  # noqa
+        return Response(serializer.data)
```

Comment:
serializer_retrieve, it's for retrieve action?
For list action use default serializer_class